### ISSUE TITLE
FunctionNotDefined object has no attribute 'func_name'

### ISSUE THREAD

**Issue by ComaVN** [danthedeckie/simpleeval#41]
According to [the source code](https://github.com/danthedeckie/simpleeval/blame/0.9.5/simpleeval.py#L113), the `FunctionNotDefined` exception has an attribute `func_name`, and has had it since the initial commit. However, when using `pip` to install the package, the same section looks like this:

```
class FunctionNotDefined(InvalidExpression):
    """ sorry! That function isn't defined! """
    def __init__(self, func_name, expression):
        self.message = "Function '{0}' not defined," \
                       " for expression '{1}'.".format(func_name, expression)
        self.__name__ = func_name
        self.expression = expression
```

so the attribute is called `__name__`.

The weird thing is, that the tar on [PyPI](https://pypi.org/project/simpleeval/#files) DOES seem to contain the correct code (no diff with the github version.

What's going on here?

To reproduce, I did:

```
docker run --rm -it python:3.7.0-stretch bash
pip install simpleeval
grep -A6 'class FunctionNotDefined' /usr/local/lib/python3.7/site-packages/simpleeval.py
```

**Comment by ComaVN** [comment]
To reproduce the actual problem (not being able to access func_name):
```
docker run --rm -i python:3.7.0-stretch <<EOT
import pip._internal as pip
pip.main(['install', 'simpleeval'])
from simpleeval import (simple_eval, FunctionNotDefined)
try:
    simple_eval('doesnotexist()')
except FunctionNotDefined as e:
    print(e)
    print(e.func_name)
EOT
```

result:
```
Collecting simpleeval
(...)
Successfully installed simpleeval-0.9.5
Function 'doesnotexist' not defined, for expression 'doesnotexist()'.
Traceback (most recent call last):
(...)
During handling of the above exception, another exception occurred:
(...)
AttributeError: 'FunctionNotDefined' object has no attribute 'func_name'
```

I expected to just get the function name of the undefined function.

**Comment by danthedeckie** [comment]
Oh that's weird.  I guess it must be a 2to3 or similar conversion tool problem...
Probably lib2to3/fixes/fix_funcattrs.py getting confused.

Please could you confirm that this is fixed with 0.9.6?  I've changed the way it works now to use setattr rather than directly .func_name, which I think should bypass 2to3.

I added a couple test cases too for this issue.
